docs/DeveloperGuide.adoc

```diff
@@ -281,37 +281,74 @@ image::CloseCommitActivityDiagram.png[]
+The following activity diagram summarizes what happens when a user executes a new command:
 image::ModeCommandActivity.png[]
 
 ==== Design Considerations
-===== Aspect: How Mode executes
-* **Current choice:** Toggles between mode through a boolean value.
-** Pros: Easily implemented.
-** Cons: An additional variable to check when executing other commands. Possibility of incorrect manipulation of an Eatery object.
+===== Aspect: How Mode execute
+* **Alternative 1 (current choice):** Toggle between mode through a boolean value.
+** Pros: Easy for implementation.
+** Cons: An additional factor to check when executing any other commands; Possibility of mis-manipulation of data.
 
 ===== Aspect: Data structure to support Mode Command
-* **Current choice:** Maintains two separate lists of Eateries for Main mode and Todo Mode.
-** Pros: Todo Eateries stored apart from Main Eateries. Commands executed will only affect Eateries stored in a particular list.
-** Cons: Requires proper handling of individual data structures to ensure each list is maintained and updated correctly.
+* **Alternative 1 (current choice):** Maintain two separate lists for Main mode and Todo Mode.
+** Pros: Data between the two modes will be separated apart. Commands executed will only affect data stored in the list for the particular mode.
+** Cons: More effort required for maintenance purposes. Need to make sure that data from main list should not go into todo list, and vice versa
 //end::modeCommand[]
 
+//tag::saveTodoCommand[]
+=== Saving Todo Eatery Feature
+==== Implementation
+This command allows the user to save an eatery from the todo list to the main eatery list. It extends `Command`.
+
+Given below is an example usage of how the SaveTodo Command behaves.
+
+Step 1: The user launches the application. Data from EateryList will be fetched and initialised in main mode by default.
+
+Step 2: The user executes mode command and the mode of the application will be switched to Todo mode, displaying items in the todo list instead.
+
+Step 3: The user can execute save 1, indicating to the system the index of eatery to be saved.
+
+Step 4: The eatery in the todo list will be removed and an add command will be constructed in the CommandBox for the user to input any necessary fields before adding.
+
+The following sequence diagram shows how the `SaveTodo` command works:
+image::SaveTodoCommandSequence.png[]
+
+The following activity diagram summarizes what happens when a user executes a new command:
+image::SaveTodoCommandActivity.png[]
+
+==== Design Considerations
+===== Aspect: How SaveTodo execute
+* **Alternative 1 (current choice):** Construct the commands as a string before displaying to the user.
+** Pros: User-friendly. User will be able to add the eatery to the main list with minimum effort.
+** Cons: Unable to save directly to the main list from todo list. Possible situation can be that the user did not `add` the new eatery after executing `saveTodo` command.
```

Comment:
Possible situation can be that the user did not add the new eatery after executing `saveTodo` command. <- don't need to format the "add"